### PR TITLE
MWPW-125561 - UAR Results Block

### PR DESCRIPTION
**Pull request by colloyd**
* Initial commit
* Basic UAR results
* Nested UAR results
* Style support

Resolves: [MWPW-125561](https://jira.corp.adobe.com/browse/MWPW-125561)

**Test URLs:**
- Before: https://main--milo--adobecom.hlx.page/drafts/colloyd/uar-results-block/uar-results?quizKey=cc-uar-result&martech=off
- After: https://mwpw-125561-uar-results-block--milo--colloyd.hlx.page/drafts/colloyd/uar-results-block/uar-results?quizKey=cc-uar-result&martech=off

### CODE REVIEW

**Comment by colloyd** [comment]
To properly test this, hit the test url and open dev tools. Add the 'cc-uar-result' key to Local Storage and set the value to this:

`{"structure-fragments":["https://main--milo--adobecom.hlx.page/fragments/colloyd/sample-uar-fragments/uar-sample-marquee-cc","https://main--milo--adobecom.hlx.page/fragments/colloyd/sample-uar-fragments/uar-sample-card-list"],"marquee-product":["https://main--milo--adobecom.hlx.page/fragments/colloyd/sample-uar-fragments/uar-sample-marquee-product-au","https://main--milo--adobecom.hlx.page/fragments/colloyd/sample-uar-fragments/uar-sample-marquee-product-lr","https://main--milo--adobecom.hlx.page/fragments/colloyd/sample-uar-fragments/uar-sample-marquee-product-id"],"commerce-card":["https://main--milo--adobecom.hlx.page/fragments/colloyd/sample-uar-fragments/uar-sample-card-au","https://main--milo--adobecom.hlx.page/fragments/colloyd/sample-uar-fragments/uar-sample-card-lr","https://main--milo--adobecom.hlx.page/fragments/colloyd/sample-uar-fragments/uar-sample-card-id"]}`

**Review by echampio-at-adobe — APPROVED** [review]
It's a nice validation of Milo as a framework that so much can be accomplished with so little code. In the long run, might be nice to add a real simple regex to validate the fragment paths, just to make sure they're on the same domain under the fragments path. I'm sure createFragment will do it's own validation. Just don't like to propagate garbage. Might also be nice to add a logger method of our own that can use the console while we're banging away among our selves, but that we can later redirect to lana without changing a bunch of lines everywhere. These comments are just future ideas, though.

**Review by JackySun9 — APPROVED** [review]

**Comment by echampio-at-adobe** [comment]
> I do see the pages with content after I setup local storage key, just want to know is it dynamic to generate marquee-product and commerce-card pages according to some excel sheets?

Hopefully, it will be. The list of fragments you see in local storage should eventually be based upon the matches made in results.xslx, Thus, dynamic.

File: libs/blocks/uar-results/uar-results.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    const a = createTag('a', { href });
+    el.append(a);
+    const { default: createFragment } = await import('../fragment/fragment.js');
+    await createFragment(a);
+  }
+}
+
+function redirectPage(quizUrl) {
+  const url = (quizUrl) ? quizUrl.text : 'https://adobe.com';
+  console.log(`Invalid Result, redirecting to: ${url}`);
```
Comment (rgclayton) [inline libs/blocks/uar-results/uar-results.js:17]: Should tap into Lana

Comment (echampio-at-adobe) [reply libs/blocks/uar-results/uar-results.js:17]: Hi @rgclayton. I mentioned lana in my comment too. I asked whether we should just define our own log wrapper so we can use the console for now, but send everything to lana when we're really ready to merge to main. Has anyone else written a method like that already that we could import?

**Review by echampio-at-adobe — COMMENTED** [review]
Hi @colloyd,

Given your code, how will it be possible for a section like "More apps to explore" to drop out when there aren't any cards specified?

https://www.adobe.com/creativecloud/quiz-recommender-var/result.html#type=cc:app-reco&quiz=UARv21&selectedOptions=category_video/Flagship/Product_AN_Video/plan_personal
Screenshot 2023-04-13 at 2 26 08 PM

**Comment by rgclayton** [comment]
> In the long run, might be nice to add a real simple regex to validate the fragment paths, just to make sure they're on the same domain under the fragments path. I'm sure createFragment will do it's own validation.

I think you can use `getConfig()` from `utils.js`.
[Here's an example ](https://github.com/adobecom/milo/blob/main/libs/blocks/carousel/carousel.js#L1-L9)

@colloyd @echampio-at-adobe

File: libs/blocks/uar-results/uar-results.js (hunk 1 above)
Comment (rgclayton) [reply libs/blocks/uar-results/uar-results.js:17]: > Has anyone else written a method like that already that we could import?

@echampio-at-adobe not that i know of.

Comment (colloyd) [reply libs/blocks/uar-results/uar-results.js:17]: That's the plan, I don't want to have redirects happening when we're debugging all of this, hence the console.log()

**Pull request merged**